### PR TITLE
Add purge cron job for Dovecot dbox format

### PR DESCRIPTION
**Pull request by gmasse**
Related to feature request #1697

### CODE REVIEW

File: target/start-mailserver.sh [hunk 1]
```diff
@@ -543,13 +543,25 @@ function _setup_dovecot
 
   # set mail_location according to mailbox format
   case "${DOVECOT_MAILBOX_FORMAT}" in
-    sdbox|mdbox|maildir )
+    sdbox|mdbox )
```
Comment (erik-wramner) [inline target/start-mailserver.sh:546]: Why has maildir been removed here? The cron job is not needed for that, but how about the other changes?

File: target/start-mailserver.sh [hunk 2]
```diff
@@ -543,13 +543,25 @@ function _setup_dovecot
...
       _notify 'inf' "Dovecot ${DOVECOT_MAILBOX_FORMAT} format configured"
       sed -i -e 's/^mail_location = .*$/mail_location = '"${DOVECOT_MAILBOX_FORMAT}"':\/var\/mail\/%d\/%n/g' /etc/dovecot/conf.d/10-mail.conf
+
+      _notify 'inf' "Creating cron job for dbox purge"
+      cat <<EOF >/etc/cron.d/dovecot-purge
+#
```
Comment (erik-wramner) [inline target/start-mailserver.sh:552]: This works, but personally I would prefer to keep the file separate and just make a copy here. That keeps the start script a bit simpler and makes it easier to change the cronjob.

**Comment by erik-wramner** [comment]
The unit tests failed on quota. Not sure why, but it is not one of the common failures. Please push again to re-trigger the tests so that we can see if it was coincidence or an error. Perhaps related to the maildir change?

File: target/start-mailserver.sh (hunk 1 above)
Comment (gmasse) [reply target/start-mailserver.sh:546]: maildir is the default behavior

File: target/start-mailserver.sh (hunk 2 above)
Comment (gmasse) [reply target/start-mailserver.sh:552]: Where is the best place to put a crontab file? It could be in `target/dovecot/`.

**Comment by aendeavor** [comment]
@erik-wramner seems like to quota test still is somewhat random, yes - I had hoped to not see this again. But at the current failure rate, this randomness is somewhat acceptable and way better than a few weeks ago.

@gmasse I agree with what @erik-wramner has proposed. Nice job:)

File: target/start-mailserver.sh (hunk 2 above)
Comment (erik-wramner) [reply target/start-mailserver.sh:552]: Sounds good to me.

**Comment by aendeavor** [comment]
@gmasse could you do a dummy-commit to re-trigger the tests?

**Comment by gmasse** [comment]
@aendeavor, I will push a new commit integrating @erik-wramner advice.

**Review by erik-wramner — APPROVED** [review]

**Review by aendeavor — APPROVED** [review]

**Pull request merged**